io/src/lib.rs

```diff
@@ -75,6 +107,15 @@ impl<R: std::io::Read> Read for R {
     }
 }
 
+#[cfg(feature = "std")]
+impl<R: std::io::BufRead + Read + ?Sized> BufRead for R {
+    #[inline]
+    fn fill_buf(&mut self) -> Result<&[u8]> { Ok(self.fill_buf()?) }
+
+    #[inline]
+    fn consume(&mut self, amount: usize) { self.consume(amount) }
```

Comment:
How do we know it's not recursion? Use UFCS to disambiguate?

Reply:
I tried both originally and thought the compiler must be able to figure it out. Anyways, used UFCS now.